src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -31,44 +32,177 @@ class SerializerPass implements CompilerPassInterface
+                }
+
+                unset($tag['serializer'], $tag['built-in']);
+
+                foreach ($names as $name) {
+                    $definition->addTag($tagName.'.'.$name, $tag);
+                }
             }
+        }
+    }
 
-            $container->getParameterBag()->remove('serializer.default_context');
+    private function bindDefaultContext(ContainerBuilder $container, array $services, array $defaultContext): void
+    {
+        foreach ($services as $id) {
+            $definition = $container->getDefinition((string) $id);
+            $definition->setBindings(['array $defaultContext' => new BoundArgument($defaultContext, false)] + $definition->getBindings());
         }
+    }
 
+    private function configureSerializer(ContainerBuilder $container, string $id, array $normalizers, array $encoders, string $serializerName): void
+    {
         if ($container->getParameter('kernel.debug') && $container->hasDefinition('serializer.data_collector')) {
             foreach ($normalizers as $i => $normalizer) {
                 $normalizers[$i] = $container->register('.debug.serializer.normalizer.'.$normalizer, TraceableNormalizer::class)
-                    ->setArguments([$normalizer, new Reference('serializer.data_collector')]);
+                    ->setArguments([$normalizer, new Reference('serializer.data_collector'), $serializerName]);
             }
 
             foreach ($encoders as $i => $encoder) {
                 $encoders[$i] = $container->register('.debug.serializer.encoder.'.$encoder, TraceableEncoder::class)
-                    ->setArguments([$encoder, new Reference('serializer.data_collector')]);
+                    ->setArguments([$encoder, new Reference('serializer.data_collector'), $serializerName]);
             }
         }
 
-        $serializerDefinition = $container->getDefinition('serializer');
+        $serializerDefinition = $container->getDefinition($id);
         $serializerDefinition->replaceArgument(0, $normalizers);
         $serializerDefinition->replaceArgument(1, $encoders);
     }
+
+    private function configureNamedSerializers(ContainerBuilder $container): void
+    {
+        $defaultSerializerNameConverter = $container->hasParameter('.serializer.name_converter')
+            ? $container->getParameter('.serializer.name_converter') : null;
+
+        foreach ($container->getParameter('.serializer.named_serializers') as $serializerName => $config) {
+            $config += ['default_context' => [], 'name_converter' => null];
+            $serializerId = 'serializer.'.$serializerName;
+
+            if (!$normalizers = $this->findAndSortTaggedServices('serializer.normalizer.'.$serializerName, $container)) {
+                throw new RuntimeException(\sprintf('You must tag at least one service as "serializer.normalizer" with the name "%s" to use the "%s" service.', $serializerName, $serializerId));
```

Comment:
The message is not very actionable given `with the name`. We are rather looking for a `serializer` tag attribute right?
 
```suggestion
                throw new RuntimeException(\sprintf('Named serializer "%1$s" requires to have at least one normalizer registered. Tag them as "serializer.normalizer" with "serializer" attribute set to "%1$s".', $serializerName));
```
Maybe?

Reply:
Done https://github.com/symfony/symfony/compare/9e76b020d77ccb4d42ef0c00598221598e1e28e5..7d850a081dbe00c0cb3050c12b53e44376aa9887